primal-bit/src/lib.rs

```diff
@@ -171,11 +170,21 @@ impl BitVec {
     /// }
     /// ```
     pub fn from_elem(nbits: usize, bit: bool) -> BitVec {
+        use std::ptr;
+
         let nblocks = nbits.checked_add(BITS - 1).expect("capacity overflow") / BITS;
+        let mut out_vec = Vec::with_capacity(nblocks);
+
+        unsafe {
+            out_vec.set_len(nblocks);
+            ptr::write_bytes(out_vec.as_mut_ptr(), if bit { !0 } else { 0 }, nblocks);
+        }
```

Comment:
[Looks like you're absolutely right](https://github.com/rust-lang/rust/blob/b8499e0460e36dfb304b8800c83587b2cacc887c/src/liballoc/vec.rs#L1462-L1478). This is the way to go, for sure. I'll get on it.